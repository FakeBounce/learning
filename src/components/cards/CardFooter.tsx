import { Trans } from '@lingui/macro';
import { Box } from '@mui/material';
import ActionButton from '@src/components/lms/ActionButton';
import { memo } from 'react';

interface CardFooterProps {
  isLoading: boolean;
  cancelAction: () => void;
}

function CardFooter({ isLoading, cancelAction }: CardFooterProps) {
  return (
    <Box display="flex" gap={2}>
      <ActionButton actionType="cancel" loading={isLoading} onClick={cancelAction}>
        <Trans>Annuler</Trans>
      </ActionButton>
      <ActionButton type="submit" role="submit" loading={isLoading}>
        <Trans>Enregistrer</Trans>
      </ActionButton>
    </Box>
  );
}

export default memo(CardFooter);
