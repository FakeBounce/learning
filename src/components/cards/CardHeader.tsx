import { Box, Typography } from '@mui/material';
import { memo, ReactNode } from 'react';
import ActionButton from '@src/components/lms/ActionButton';

interface CardHeaderActionProps {
  action: () => void | VoidFunction;
  actionType?: 'cancel' | 'update' | 'action';
  actionText: ReactNode;
}

interface CardHeaderProps {
  headerText: ReactNode;
  headerColor?: string;
  actions?: CardHeaderActionProps[] | null;
}

function CardHeader({ headerText, headerColor, actions = null }: CardHeaderProps) {
  return (
    <Box
      display="flex"
      gap={2}
      flexDirection={[actions && actions.length > 1 ? 'column' : 'row', 'row']}
    >
      <Typography
        sx={{
          fontSize: (theme) => theme.typography.h3.fontSize,
          fontWeight: (theme) => theme.typography.fontWeightBold,
          color: (theme) => headerColor ?? theme.palette.grey[800]
        }}
      >
        {headerText}
      </Typography>
      {actions && (
        <Box display="flex" gap={2}>
          {actions.map((action, index) => (
            <ActionButton
              key={index}
              actionType={action.actionType ?? 'action'}
              sx={{ textTransform: 'none' }}
              onClick={action.action}
            >
              {action.actionText}
            </ActionButton>
          ))}
        </Box>
      )}
    </Box>
  );
}

export default memo(CardHeader);
