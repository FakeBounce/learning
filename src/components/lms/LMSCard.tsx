import { Stack, Paper, Box } from '@mui/material';
import { ReactNode } from 'react';

interface LMSCardProps {
  cardCss?: Record<string, any>;
  isPageCard?: boolean;
  canExpand?: boolean;
  header?: ReactNode;
  footer?: ReactNode;
  children: ReactNode;
  contentPadding?: number;
}

const defaultCss = {
  display: 'flex',
  flexDirection: 'column',
  flex: '1 1 0',
  maxWidth: 500,
  paddingX: 5,
  paddingY: 4,
  borderColor: 'transparent',
  borderRadius: 4,
};

const pageCardCss = {
  boxSizing: 'border-box',
  display: 'flex',
  flexDirection: 'column',
  flex: '1 1 0',
  borderColor: 'transparent',
  borderRadius: 4,
  padding: 0,
  maxWidth: '100%',
  paddingBottom: 3,
  marginBottom: 2
};

export default function LMSCard({
  cardCss,
  isPageCard = false,
  canExpand = false,
  header,
  footer,
  children,
  contentPadding = 4
}: LMSCardProps) {
  // Keep theme and matches in case need later
  // const theme = useTheme();
  // const matches = useMediaQuery(theme.breakpoints.up('sm'));

  const chosenCss = isPageCard ? pageCardCss : defaultCss;
  const mergedCardCss = {
    ...chosenCss,
    ...cardCss
  };

  if (isPageCard) {
    return (
      <Stack px={2} maxWidth="100%" flex={'1 1 0'}>
        <Paper elevation={3} sx={{ ...mergedCardCss }}>
          {header && (
            <Stack spacing={2} p={2} px={4}>
              {header}
            </Stack>
          )}
          {canExpand ? (
            <Box gap={2} px={contentPadding} sx={{ flex: '1 1 0' }}>
              {children}
            </Box>
          ) : (
            <Stack spacing={2} px={contentPadding} sx={{ flex: '1 1 0' }}>
              {children}
            </Stack>
          )}

          {footer && (
            <Stack spacing={2} pt={2} px={4} sx={{ mt: 2 }}>
              {footer}
            </Stack>
          )}
        </Paper>
      </Stack>
    );
  }

  return (
    <Paper elevation={3} sx={{ ...mergedCardCss }}>
      {header && (
        <Stack spacing={2} mb={2}>
          {header}
        </Stack>
      )}
      {children}
      {footer && (
        <Stack spacing={2} sx={{ mt: 2 }}>
          {footer}
        </Stack>
      )}
    </Paper>
  );
}
