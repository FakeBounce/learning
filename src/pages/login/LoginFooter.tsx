// @mui
import { LoadingButton } from '@mui/lab';
import { useLocales } from '@src/locales';
import theme from '@theme';

// ----------------------------------------------------------------------

export default function LoginFooter({ isLoading }: { isLoading: boolean }) {
  const { translate } = useLocales();

  return (
    <LoadingButton
      fullWidth
      size="large"
      type="submit"
      loading={isLoading}
      sx={{
        borderRadius: 2,
        bgcolor: theme.palette.green[700],
        color: 'white',
        '&:hover': {
          bgcolor: theme.palette.green[800]
        }
      }}
    >
      {translate('LOGIN_SUBMIT')}
    </LoadingButton>
  );
}
