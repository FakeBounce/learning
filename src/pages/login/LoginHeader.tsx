import { Trans } from '@lingui/macro';
import { Typography, Box } from '@mui/material';
import { NavLink } from 'react-router-dom';
import { Theme } from '@mui/material/styles';

export default function LoginHeader() {
  return (
    <>
      <Typography
        variant="h3"
        fontWeight={(theme) => theme.typography.fontWeightMedium}
        fontSize={(theme) => theme.typography.h3.fontSize}
      >
        <Trans>Connexion</Trans>
      </Typography>
      <Box display="flex" flexDirection={['column', 'row']} mb={4}>
        <Typography fontSize={(theme) => theme.typography.body1.fontSize}>
          <Trans>Pas encore de compte ?</Trans>
        </Typography>
        <Typography
          ml={[0, 1]}
          fontWeight={(theme) => theme.typography.fontWeightBold}
          fontSize={(theme) => theme.typography.body1.fontSize}
          color={(theme) => theme.palette.primary.dark}
        >
          <Box
            component={NavLink}
            to="/register"
            sx={{ color: (theme: Theme) => theme.palette.primary.dark }}
          >
            <Trans>Cr√©er un compte</Trans>
          </Box>
        </Typography>
      </Box>
    </>
  );
}
