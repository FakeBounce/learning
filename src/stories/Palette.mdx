import { Meta, ColorPalette, ColorItem } from '@storybook/blocks';
import palette from '../theme/palette';

export const customPalette = palette();

<Meta title="Colors" />

<ColorPalette>
  <ColorItem
    title="Greyscale"
    subtitle="Some of the greys"
    colors={{
      '0': customPalette.grey[0],
      '100': customPalette.grey[100],
      '200': customPalette.grey[200],
      '300': customPalette.grey[300],
      '400': customPalette.grey[400],
      '500': customPalette.grey[500],
      '600': customPalette.grey[600],
      '700': customPalette.grey[700],
      '800': customPalette.grey[800],
      '900': customPalette.grey[900]
    }}
  />
  <ColorItem
    title="Primary"
    subtitle="Coral"
    colors={{
      Lighter: customPalette.primary.lighter,
      Light: customPalette.primary.light,
      Main: customPalette.primary.main,
      Dark: customPalette.primary.dark,
      Darker: customPalette.primary.darker,
      ContrastText: customPalette.primary.contrastText
    }}
  />
  <ColorItem
    title="Secondary"
    subtitle="Ocean"
    colors={{
      Lighter: customPalette.secondary.lighter,
      Light: customPalette.secondary.light,
      Main: customPalette.secondary.main,
      Dark: customPalette.secondary.dark,
      Darker: customPalette.secondary.darker,
      ContrastText: customPalette.secondary.contrastText
    }}
  />
  <ColorItem
    title="Info"
    subtitle="Info"
    colors={{
      Lighter: customPalette.info.lighter,
      Light: customPalette.info.light,
      Main: customPalette.info.main,
      Dark: customPalette.info.dark,
      Darker: customPalette.info.darker,
      ContrastText: customPalette.info.contrastText
    }}
  />
  <ColorItem
    title="Success"
    subtitle="Green"
    colors={{
      Lighter: customPalette.success.lighter,
      Light: customPalette.success.light,
      Main: customPalette.success.main,
      Dark: customPalette.success.dark,
      Darker: customPalette.success.darker,
      ContrastText: customPalette.success.contrastText
    }}
  />
  <ColorItem
    title="Warning"
    subtitle="Orange"
    colors={{
      Lighter: customPalette.warning.lighter,
      Light: customPalette.warning.light,
      Main: customPalette.warning.main,
      Dark: customPalette.warning.dark,
      Darker: customPalette.warning.darker,
      ContrastText: customPalette.warning.contrastText
    }}
  />
  <ColorItem
    title="Error"
    subtitle="Red"
    colors={{
      Lighter: customPalette.error.lighter,
      Light: customPalette.error.light,
      Main: customPalette.error.main,
      Dark: customPalette.error.dark,
      Darker: customPalette.error.darker,
      ContrastText: customPalette.error.contrastText
    }}
  />
</ColorPalette>
