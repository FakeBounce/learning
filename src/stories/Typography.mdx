import { Meta, Typeset } from '@storybook/blocks';
import typography from '../theme/typography';

<Meta title="Typography" />

export const SampleText = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.';

# Typography

## Balises h1->h6

**Font:** Public Sans, sans-serif

**Weights:** 400(regular)

<Typeset
  fontSizes={[
    typography.h1.fontSize,
    typography.h2.fontSize,
    typography.h3.fontSize,
    typography.h4.fontSize,
    typography.h5.fontSize,
    typography.h6.fontSize
  ]}
  fontWeight={typography.h1.fontWeight}
  sampleText={SampleText}
  fontFamily={typography.fontFamily}
/>

## Sous-titre 1 et 2

**Font:** Public Sans, sans-serif

**Weights:** 400(regular)

<Typeset
  fontSizes={[typography.subtitle1.fontSize, typography.subtitle2.fontSize]}
  fontWeight={typography.h1.fontWeight}
  sampleText={SampleText}
  fontFamily={typography.fontFamily}
/>

## Corps de texte 1 et 2 dans diff√©rents poids

**Font:** Public Sans, sans-serif

**Weights:** 400(regular)

<Typeset
  fontSizes={[typography.body1.fontSize, typography.body2.fontSize]}
  fontWeight={typography.h1.fontWeight}
  sampleText={SampleText}
  fontFamily={typography.fontFamily}
/>

**Font:** Public Sans, sans-serif

**Weights:** 600(medium)

<Typeset
  fontSizes={[typography.body1.fontSize, typography.body2.fontSize]}
  fontWeight={typography.fontWeightMedium}
  sampleText={SampleText}
  fontFamily={typography.fontFamily}
/>

**Font:** Public Sans, sans-serif

**Weights:** 700(bold)

<Typeset
  fontSizes={[typography.body1.fontSize, typography.body2.fontSize]}
  fontWeight={typography.fontWeightBold}
  sampleText={SampleText}
  fontFamily={typography.fontFamily}
/>

## Caption, overline et bouton

**Font:** Public Sans, sans-serif

**Weights:** 400(regular)

<Typeset
  fontSizes={[typography.caption.fontSize]}
  fontWeight={typography.h1.fontWeight}
  sampleText={SampleText}
  fontFamily={typography.fontFamily}
/>

**Font:** Public Sans, sans-serif

**Weights:** 400(regular)

<Typeset
  fontSizes={[typography.overline.fontSize]}
  fontWeight={typography.h1.fontWeight}
  sampleText={SampleText}
  fontFamily={typography.fontFamily}
/>

**Font:** Public Sans, sans-serif

**Weights:** 400(regular)

<Typeset
  fontSizes={[typography.button.fontSize]}
  fontWeight={typography.h1.fontWeight}
  sampleText={SampleText}
  fontFamily={typography.fontFamily}
/>
