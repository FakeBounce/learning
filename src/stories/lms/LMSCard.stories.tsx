import type { Meta, StoryObj } from '@storybook/react';
import LMSCard from '@src/components/lms/LMSCard';
import Typography from '@mui/material/Typography';
import * as CardFooterStories from '@src/stories/cards/CardFooter.stories';
import CardFooter from '@src/components/cards/CardFooter';
import LoginFooter from '@src/pages/login/LoginFooter';
import LoginHeader from '@src/pages/login/LoginHeader';
import CardHeader from '@src/components/cards/CardHeader';
import * as CardHeaderStories from '@src/stories/cards/CardHeader.stories';

const meta: Meta = {
  title: 'Example/LMSCard',
  component: LMSCard,
  tags: ['autodocs'],
  parameters: {
    layout: 'centered'
  }
};

export default meta;

type Story = StoryObj<typeof LMSCard>;

export const Default: Story = {
  args: {
    children: <Typography>This is a default LMSCard without header or footer.</Typography>
  }
};

export const WithHeaderAndFooter: Story = {
  args: {
    header: <LoginHeader />,
    footer: <LoginFooter isLoading={false} />,
    children: <Typography>This is an LMSCard with header and footer. (Login style)</Typography>
  }
};

export const PageCard: Story = {
  args: {
    isPageCard: true,
    header: <CardHeader {...CardHeaderStories.WithSingleAction.args} />,
    footer: <CardFooter {...CardFooterStories.Default.args} />,
    children: (
      <Typography>This is a page style LMSCard with header and footer. (Grid style)</Typography>
    )
  }
};
