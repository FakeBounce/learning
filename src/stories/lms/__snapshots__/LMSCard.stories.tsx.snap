// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example/LMSCard Default smoke-test 1`] = `
<div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1slpo8t-MuiPaper-root">
  <p class="MuiTypography-root MuiTypography-body1 css-gkbk35-MuiTypography-root">
    This is a default LMSCard without header or footer.
  </p>
</div>
`;

exports[`Example/LMSCard PageCard smoke-test 1`] = `
<div class="MuiStack-root css-dnja1o-MuiStack-root">
  <div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-8c9seq-MuiPaper-root">
    <div class="MuiStack-root css-1a5lnx6-MuiStack-root">
      <div class="MuiBox-root css-10egq61">
        <p class="MuiTypography-root MuiTypography-body1 css-tjb74-MuiTypography-root">
          Header with single action
        </p>
        <button class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-121cmne-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
                tabindex="0"
                type="button"
                id=":r1:"
        >
          Action
          <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
          </span>
        </button>
      </div>
    </div>
    <div class="MuiStack-root css-141v11l-MuiStack-root">
      <p class="MuiTypography-root MuiTypography-body1 css-gkbk35-MuiTypography-root">
        This is a page style LMSCard with header and footer. (Grid style)
      </p>
    </div>
    <div class="MuiStack-root css-96ys31-MuiStack-root">
      <div class="MuiBox-root css-10egq61">
        <button class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1nvwnbn-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
                tabindex="0"
                type="button"
                id=":r2:"
        >
          Annuler
          <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
          </span>
        </button>
        <button class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-121cmne-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
                tabindex="0"
                type="submit"
                id=":r3:"
                role="submit"
        >
          Enregistrer
          <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Example/LMSCard WithHeaderAndFooter smoke-test 1`] = `
<div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1slpo8t-MuiPaper-root">
  <div class="MuiStack-root css-1saxivv-MuiStack-root">
    <h3 class="MuiTypography-root MuiTypography-h3 css-qtlqwc-MuiTypography-root">
      Connexion
    </h3>
    <div class="MuiBox-root css-1cg1idb">
      <p class="MuiTypography-root MuiTypography-body1 css-a2cwj1-MuiTypography-root">
        Pas encore de compte ?
      </p>
      <p class="MuiTypography-root MuiTypography-body1 css-ls5w3c-MuiTypography-root">
        <a class="MuiBox-root css-170kp7e"
           href="/register"
        >
          Cr√©er un compte
        </a>
      </p>
    </div>
  </div>
  <p class="MuiTypography-root MuiTypography-body1 css-gkbk35-MuiTypography-root">
    This is an LMSCard with header and footer. (Login style)
  </p>
  <div class="MuiStack-root css-11zvfn9-MuiStack-root">
    <button class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge MuiButton-fullWidth MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge MuiButton-fullWidth css-1msqpia-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
            tabindex="0"
            type="submit"
            id=":r0:"
            role="submit"
    >
      Se connecter
      <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
      </span>
    </button>
  </div>
</div>
`;
